### ISSUE TITLE
Doesn't save resume position when casting

### ISSUE THREAD

**Issue by Luna712** [recloudstream/cloudstream#804]
### Steps to reproduce

1. Cast a video
2. Fast forward to a point
3. Exit casting

### Expected behavior

When you resume either normally on phone or casting again position should resume

### Actual behavior

It doesn't resume

### Cloudstream version and commit hash

4.2.1 530619c

### Android version

Android 14

### Logcat

Whenever I try to open logcat the app crashes. This happens on previous Android versions as well.

### Other details

Also another issue seems to happen since Android 14 (Didn't notice before but may have happened before) when watching a series and you complete one episode, the entire series seems to disappear from the continue watching scroller and you have search to find it again.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to pre-release version **[Latest](https://github.com/recloudstream/cloudstream/releases)**.
- [X] If related to a provider, I have checked the site and it works, but not the app.
- [X] I will fill out all of the requested information in this form.

**Comment by fire-light42** [comment]
iirc chromecasting will be reworked to support headers and other stuff by using a cast implementation from another foss app, so this problem will be fixed when that happens

**Comment by Luna712** [comment]
@fire-light42

That makes sense thanks!

Also, it seems the issue moderation action is [failing](https://github.com/recloudstream/cloudstream/actions/runs/7152640082) now

**Comment by Blatzar** [comment]
Yes, currently https://fcast.org/ is looking quite attractive, they are only missing header support, but that seems quite easily fixed given that they are foss
